lightning/src/routing/router.rs

```diff
@@ -795,184 +813,190 @@ where L::Target: Logger {
 					let over_path_minimum_msat = amount_to_transfer_over_msat >= $directional_info.htlc_minimum_msat &&
 						amount_to_transfer_over_msat >= $next_hops_path_htlc_minimum_msat;
 
-					// If HTLC minimum is larger than the amount we're going to transfer, we shouldn't
-					// bother considering this channel.
-					// Since we're choosing amount_to_transfer_over_msat as maximum possible, it can
-					// be only reduced later (not increased), so this channel should just be skipped
-					// as not sufficient.
-					if !over_path_minimum_msat {
-						hit_minimum_limit = true;
-					} else if contributes_sufficient_value {
-						// Note that low contribution here (limited by available_liquidity_msat)
-						// might violate htlc_minimum_msat on the hops which are next along the
-						// payment path (upstream to the payee). To avoid that, we recompute
-						// path fees knowing the final path contribution after constructing it.
-						let path_htlc_minimum_msat = compute_fees($next_hops_path_htlc_minimum_msat, $directional_info.fees)
-							.and_then(|fee_msat| fee_msat.checked_add($next_hops_path_htlc_minimum_msat))
-							.map(|fee_msat| cmp::max(fee_msat, $directional_info.htlc_minimum_msat))
-							.unwrap_or_else(|| u64::max_value());
-						let hm_entry = dist.entry($src_node_id);
-						let old_entry = hm_entry.or_insert_with(|| {
-							// If there was previously no known way to access
-							// the source node (recall it goes payee-to-payer) of $chan_id, first add
-							// a semi-dummy record just to compute the fees to reach the source node.
-							// This will affect our decision on selecting $chan_id
-							// as a way to reach the $dest_node_id.
-							let mut fee_base_msat = u32::max_value();
-							let mut fee_proportional_millionths = u32::max_value();
-							if let Some(Some(fees)) = network_nodes.get(&$src_node_id).map(|node| node.lowest_inbound_channel_fees) {
-								fee_base_msat = fees.base_msat;
-								fee_proportional_millionths = fees.proportional_millionths;
-							}
-							PathBuildingHop {
-								node_id: $dest_node_id.clone(),
-								short_channel_id: 0,
-								channel_features: $chan_features,
-								fee_msat: 0,
-								cltv_expiry_delta: 0,
-								src_lowest_inbound_fees: RoutingFees {
-									base_msat: fee_base_msat,
-									proportional_millionths: fee_proportional_millionths,
-								},
-								channel_fees: $directional_info.fees,
-								next_hops_fee_msat: u64::max_value(),
-								hop_use_fee_msat: u64::max_value(),
-								total_fee_msat: u64::max_value(),
-								htlc_minimum_msat: $directional_info.htlc_minimum_msat,
-								path_htlc_minimum_msat,
-								path_penalty_msat: u64::max_value(),
-								was_processed: false,
-								#[cfg(any(test, feature = "fuzztarget"))]
-								value_contribution_msat,
+					if let Some(hop_total_cltv_delta) = ($next_hops_cltv_delta as u32).checked_add($directional_info.cltv_expiry_delta as u32) {
```

Comment:
I'm touching a lot of this code in #1227. To avoid merge conflicts, could we do something like the following to keep the patch set minimal?

```rust
let hop_total_cltv_delta = ($next_hops_cltv_delta as u32)
	.checked_add($directional_info.cltv_expiry_delta as u32))
	.unwrap_or(u32::max_value());
```

Reply:
Sure! I changed it (back) with https://github.com/lightningdevkit/rust-lightning/pull/1234/commits/367ed31dc980a099a5aa963a09c953ff80bbacd7.